src/CodeGenerator/src/Definition/UnionShape.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\CodeGenerator\Definition;
+
+/**
+ * @internal
+ */
+class UnionShape extends StructureShape
```

Comment:
should we actually extend StructureShape ? Many places in the code generator rely on `instanceof` to choose their behavior, and they throw for unsupported shapes. Should union shapes be treated like normal structure shapes for most of those places ? If no, it would be better to have a separate type.